molgenis-data-index/src/main/java/org/molgenis/data/index/IndexDependencyModel.java

```diff
@@ -0,0 +1,108 @@
+package org.molgenis.data.index;
+
+import com.google.common.collect.ImmutableSet;
+import org.molgenis.data.meta.model.Attribute;
+import org.molgenis.data.meta.model.EntityType;
+import org.molgenis.util.GenericDependencyResolver;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.Maps.uniqueIndex;
+import static java.util.Collections.emptySet;
+import static java.util.stream.StreamSupport.stream;
+
+/**
+ * Models the dependencies between Entities.
+ */
+public class IndexDependencyModel
```

Comment:
Very nice! Although it's not really index specific, is it? I have a feeling this might come in handy in other places as well.